modulemd/v2/modulemd-module-stream.c

```diff
@@ -684,3 +688,53 @@ static void
 modulemd_module_stream_init (ModulemdModuleStream *self)
 {
 }
+
+
+void
+modulemd_module_stream_set_translation (ModulemdModuleStream *self,
+                                        ModulemdTranslation *translation)
+{
+  g_return_if_fail (MODULEMD_IS_MODULE_STREAM (self));
+
+  ModulemdModuleStreamPrivate *priv =
+    modulemd_module_stream_get_instance_private (self);
+
+  g_clear_pointer (&priv->translation, g_object_unref);
+  if (translation != NULL)
+    priv->translation = modulemd_translation_copy (translation);
```

Comment:
I'm really not sure we want to be copying this data around. My original thought was that we'd make these into references. In some module-stream-expansion cases, we may have MANY instances of the same module/stream pair in the Index. (Multiple versions, each with multiple contexts can grow quickly). I'd much rather we maintained a single copy at the ModuleIndex level that we associate to the individual streams via `g_object_ref()`. The summaries and descriptions can be arbitrarily long, and I'd rather not be duplicating hundreds of kilobytes of memory unnecessarily.

Also, if we move to referencing them, I think we should change the function name from `modulemd_module_stream_set_translation()` to `modulemd_module_stream_associate_translation()`, which will make it more clear that it's taking a reference and not making a copy.